Makefile

```diff
@@ -73,7 +74,8 @@ test-all-components-coverage: ## Run tests with coverage
 sync-assets: ## Sync assets folder to GCS.
 	@if [ -d "./pipelines/assets/" ] ; then \
 		echo "Syncing assets to GCS" && \
-		gsutil -m rsync -r -d ./pipelines/assets ${PIPELINE_FILES_GCS_PATH}/assets ; \
+		. ./env.sh && \
+		gsutil -m rsync -r -d ./pipelines/assets $$PIPELINE_FILES_GCS_PATH; \
```

Comment:
`$PIPELINE_FILES_GCS_PATH` should be the parent folder of the assets directory, because (in CI/CD) the compiled pipelines are copied directly to `$PIPELINE_FILES_GCS_PATH` and the contents of the assets folder are copied to the `assets` subdirectory

Reply:
I've updated the environment variable to include the /assets subdirectory to be able to construct it more easily with the `RESOURCE_SUFFIX`. I've updated in the `cloudbuild/e2e-test.yaml` as well

Reply:
This will break `release.yaml` because `training.json` and `prediction.json` are copied to `$PIPELINE_FILES_GCS_PATH` and `assets` is a directory within

I suggest you keep `assets` as a subdirectory, and instead have (in `env.sh`):

```
export PIPELINE_FILES_GCS_PATH=gs://${VERTEX_PROJECT_ID}-pl-assets/${RESOURCE_SUFFIX}
```

Reply:
It shouldn't break the `release.yaml` because the compiled pipeline jsons and assets are copied into `${_PIPELINE_PUBLISH_GCS_PATHS}`. It doesn't use the Makefile either.

Reply:
You're right. I still think you should keep it consistent though - ${_PIPELINE_PUBLISH_GCS_PATHS} in `release.yaml` is assumed not to contain `/assets` in the path, while `$PIPELINE_FILES_GCS_PATH` now **is** expected to contain `/assets` in the path

I've also realised we might need to update `terraform/modules/scheduled_pipelines/scheduled_jobs.auto.tfvars.example`

Reply:
the idea was to avoid doing `train_script_uri = f"{pipeline_files_gcs_path}/{resource_suffix}/assets/train_xgb_model.py"` by having `pipeline_files_gcs_path` fully from the environment variable.

It's also required because if the variable is not defined the above line doesn't work because path contains multiple slashes `"{pipeline_files_gcs_path}//assets/train_xgb_model.py"` which breaks the pipeline

Reply:
To update `terraform/modules/scheduled_pipelines/scheduled_jobs.auto.tfvars.example` do you mean

`pipeline_files_gcs_path = "gs://my-assets-bucket/<Git tag>"` => `pipeline_files_gcs_path = "gs://my-assets-bucket/<Git tag>/assets"`?

Reply:
In the `release.yaml` pipeline, the compiled pipelines (`training.json` and `prediction.json`) are published to `_PIPELINE_PUBLISH_GCS_PATHS`, and the `assets` folder is then a subdirectory within `_PIPELINE_PUBLISH_GCS_PATHS`

(With KFP V2 we will be publishing pipelines to Artifact Registry instead of GCS, so this won't really be a problem in the future).

I was thinking the following

In `env.sh`:

```
export RESOURCE_SUFFIX=_yourname # replace with a suffix unique to you
export PIPELINE_FILES_GCS_PATH=gs://${VERTEX_PROJECT_ID}-pl-assets/${RESOURCE_SUFFIX}
```

In `pipeline.py`:

```
@pipeline
def pipeline(
  ...
  pipeline_files_gcs_path: str = os.environ.get("PIPELINE_FILES_GCS_PATH")
  ...
):
  ...
  train_script_uri = f"{pipeline_files_gcs_path}/assets/train_xgb_model.py"
```

Reply:
(CI/CD pipelines can then be left as-is)

Reply:
I reverted PIPELINE_FILES_GCS_PATH to your suggestion. 

I made an update to the `sync-assets` make command to handle the case of empty `RESOURCE_SUFFIX` wdyt?

Reply:
I think I would put this logic into `env.sh.example` instead

```bash
PIPELINE_FILES_GCS_PATH = gs://${VERTEX_PROJECT_ID}-pl-assets
# If $RESOURCE_SUFFIX is not empty, append it to the end of $PIPELINE_FILES_GCS_PATH
if [[ ! -z "$RESOURCE_SUFFIX" ]]; then
  PIPELINE_FILES_GCS_PATH = $PIPELINE_FILES_GCS_PATH/${RESOURCE_SUFFIX}
fi
```

Reply:
That doesn't work unfortunately because the bash syntax is different to Makefile syntax (at least when using `include env.sh`). The fix would be to include Makefile syntax in env.sh.example i.e.:

```
# If $RESOURCE_SUFFIX is not empty, append it to the end of $PIPELINE_FILES_GCS_PATH
ifdef RESOURCE_SUFFIX
  export PIPELINE_FILES_GCS_PATH := $(PIPELINE_FILES_GCS_PATH)/$(RESOURCE_SUFFIX)
endif
```

I can update `sync-assets` to update the `$PIPELINE_FILES_GCS_PATH` on the fly as in your suggestion, would only add one line of code:

```
sync-assets: ## Sync assets folder to GCS.
	@if [ -d "./pipelines/assets/" ]; then \
		echo "Syncing assets to GCS"; \
		PIPELINE_FILES_GCS_PATH=$${PIPELINE_FILES_GCS_PATH}$(if $(strip $(RESOURCE_SUFFIX)),/$(RESOURCE_SUFFIX)); \
		gsutil -m rsync -r -d ./pipelines/assets "$${PIPELINE_FILES_GCS_PATH}/assets"; \
	else \
		echo "No assets folder found"; \
```

wdyt?